wcivf/apps/people/models.py

```diff
@@ -154,6 +154,22 @@ def get_max_facebook_ad_spend(self):
             )
         )
 
+    @property
+    def facebook_personal_username(self):
+        facebook_personal_url = self.facebook_personal_url
+        facebook_split = list(filter(None, facebook_personal_url.split("/")))
```

Comment:
With all these properties:

First, I wonder if there's a way to DRY up the methods a little - like pulling the main logic in to one place?


Second, parsing URLs is (or can be) hard - for example, we don't know that someone hadn't added a querystring to the end of a given URL that might look odd. 

Here's an example from the actual data: 

`https://www.facebook.com/sarahmurraylibdem?notif_id=1523864067803194&notif_t=page_invite_accept&ref=notif`. 

I think this code would reduce this down to:

`sarahmurraylibdem?notif_id=1523864067803194&notif_t=page_invite_accept&ref=notif`.

A better option might be to use Python's build in URL parsing:

```py
from urllib.parse import urlparse
parsed_url = urlparse('https://www.facebook.com/sarahmurraylibdem?notif_id=1523864067803194&notif_t=page_invite_accept&ref=notif')
```

Now `parsed_url.path` would be `sarahmurraylibdem` but might also contain more path elements, so we'd want to then do `url = parsed_url.path.split("/")[0].

Of course the _real_ solution here is to get the screen name from the 3rd party site itself, but that's a problem for another day!

Reply:
Absolutely. I'll put this on my 'things to refactor'. 👍

Reply:
Punting this to an issue as suggested so we can get this reviewed and rolled out faster.